lib/rucio/common/utils.py

```diff
@@ -1824,6 +1825,31 @@ def try_importing_policy(algorithm_type, dictionary, vo=None):
                 try_importing_policy(algorithm_type, dictionary, vo['vo'])
 
 
+def check_policy_package_version(package):
+    import importlib
+    from rucio.version import version_string
+    '''
+    Checks that the Rucio version supported by the policy package is compatible
+    with this version. Raises an exception if not.
+    :param package: the fully qualified name of the policy package
+    '''
+    try:
+        module = importlib.import_module(package)
+    except ImportError:
+        # package not found. Will be picked up elsewhere
+        return
+    if not hasattr(module, 'SUPPORTED_VERSION'):
+        # package is not versioned
+        return
+    first_compatible_version = '1.28.0'
+    supported_version = module.SUPPORTED_VERSION
+    if supported_version < first_compatible_version:
```

Comment:
Does this really do the right thing? :exploding_head: Or does it only work accidentally because it converts the version string into a float?

Reply:
It should do string comparison by default, rather than converting to float automatically. I guess the question is whether the version strings are always in the correct form for the string ordering to work as intended? It seems like they should be, and all the ones I tested worked correctly, but there may be unusual cases I haven't thought of.

Reply:
Sorry for my late reply here.
I don't think this works entirely correct, at least if you take `rcN` and `.postN` versions into account.
I am wondering in general though what the comparison here should be and I think it basically should be equivalence of the major version.

Because this code would allow if the POLICY has `SUPPORTED_VERSION=1.29` but the rucio version is already `1.30` it would not fail. Which seems wrong?

Reply:
My idea was that we could update `first_compatible_version` only when there is a change that breaks the interface used by the policy package, so a policy package would be compatible with a range of Rucio versions. But maybe that is adding too much complexity. A simple test for version equality might be better, and people can update their policy package `SUPPORTED_VERSION` field when required.

Reply:
I think this will miss an important case. New Rucio version comes out (`1.30`) which has some important API change. Thus for unmaintained policies, this will just assume it is working, while it is not.
I think it would be best to implement just a list of compatible versions. (Major versions)
If we really want to make this "elegant" we can implement something like [pip version specifiers](https://peps.python.org/pep-0440/#version-specifiers)
Where you could give a `compatible_version = ['~=1.28', '~=1.30']`etc.